lua/legendary/api/executor.lua

```diff
@@ -61,7 +61,20 @@ function M.restore_context(context, callback)
   if vim.startswith(context.mode, 'n') then
     vim.cmd('stopinsert')
   elseif Toolbox.is_visual_mode(context.mode) then
-    vim.cmd('normal! gv')
+    if true then  -- first solution
+      local vline, vcol, cline, ccol = unpack(context.marks)
+      local a, b
+      if cline < vline or ccol < vcol then  -- preserve side
+        a, b = '>', '<'
+      else
+        a, b = '<', '>'
+      end
+      vim.cmd('normal! `' .. a .. context.mode .. '`' .. b)
```

Comment:
I think I like the second solution since it's much easier to follow.

Let's remove the `if true` block and go with the 2nd solution.